### PR TITLE
[1/2] Update SDK Integration Tests to run automagicky against various data integrations

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
There are some significant test behavior changes that arise from this PR:

We finally introduce the concept of a test configuration file, located at `integration_tests/sdk/test-config.yml`. I've supplied an example format for people to copy over in `test-config-example.yml`. I imagine we'll have the full one stored in an S3 bucket somewhere.

This test configuration file will contain the server address and apikey - we'll no longer need to use environmental variables. The configuration file also contains entries for the different data integrations we will run the tests against. Every test case will run against every data integration defined in this file, unless the `--data` command line flag is supplied! 

Follow-up: Update our github actions and documentation to reflect these new changes.

## Related issue number (if any)
ENG-1837

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: integration_tests/sdk/conftest.py [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+    setup_data_integrations,
+)
 from sdk.shared import globals as test_globals
 from sdk.shared.utils import delete_flow, generate_new_flow_name
 from sdk.shared.validator import Validator
 
 
 def pytest_addoption(parser):
-    parser.addoption(f"--data", action="store", default="aqueduct_demo")
+    parser.addoption(f"--data", action="store", default=None)
```
Comment (likawind) [inline integration_tests/sdk/conftest.py:17]: why we removed the default here?

Comment (likawind) [reply integration_tests/sdk/conftest.py:17]: I think I understood now, that this flag now have controls specific data integration to use so we should follow test config file if not specified.

File: integration_tests/sdk/setup_integration.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    return data_integrations
+
+
+def get_aqueduct_config() -> Tuple[str, str]:
+    # Returns the apikey and server address.
+    test_config = _parse_config_file()
+    assert (
+        "apikey" in test_config and "address" in test_config
+    ), "apikey and address must be set in test-config.yml."
+    return test_config["apikey"], test_config["address"]
```
Comment (likawind) [inline integration_tests/sdk/setup_integration.py:192]: do you plan to use pydantic to parse this in the future?

File: integration_tests/sdk/conftest.py (hunk 1 above)
Comment (kenxu95) [reply integration_tests/sdk/conftest.py:17]: yep, exactly :)

**Review by likawind — COMMENTED** [review]
I assume you have another task for ppl to add new integrations to this suite :)

I have some questions on setup process - is the idea to have all integration start with the same set of data, so that the same tests can run consistently on them?

In this way I feel we could just require external integrations to be empty and set up by copying all data over? Just adding missing ones sounds like we need to ensure 'existing' tables to be consistent with demo. I feel this is rather hard to manage since there are many integrations we need to track.

File: integration_tests/sdk/test-config-example.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+apikey: <APIKEY>
```
Comment (hsubbaraj-spiral) [inline integration_tests/sdk/test-config-example.yml:1]: Is there a place in documentation where are the required fields for each integration are laid out?

**Review by hsubbaraj-spiral — APPROVED** [review]

File: integration_tests/sdk/setup_integration.py (hunk 2 above)
Comment (kenxu95) [reply integration_tests/sdk/setup_integration.py:192]: That's a good idea, made a task https://linear.app/aqueducthq/issue/ENG-2237/use-pydantic-for-test-config-struct-instead-of-dictionary

File: integration_tests/sdk/test-config-example.yml (hunk 3 above)
Comment (kenxu95) [reply integration_tests/sdk/test-config-example.yml:1]: For our internal dev, we should just be referencing the pydantic structs in `connect_config.py`. For users, I think adding some documentation would be good, although I don't know how much they would use something like that: https://linear.app/aqueducthq/issue/ENG-2238/add-documentation-for-connecting-integrations-from-the-sdk

**Comment by kenxu95** [comment]
@likawind 

> I assume you have another task for ppl to add new integrations to this suite :)
Yep we'll get the democratizing stuff next week.

> I have some questions on setup process - is the idea to have all integration start with the same set of data, so that the same tests can run consistently on them?
Yep, the idea is to have all integrations start with the same set of data. 


> In this way I feel we could just require external integrations to be empty and set up by copying all data over? Just adding missing ones sounds like we need to ensure 'existing' tables to be consistent with demo. I feel this is rather hard to manage since there are many integrations we need to track.
I'm not totally sure how requiring external integrations to be empty is easier to manage? If I'm running my test suite multiple times, there may be significant overhead having to populate the data each time. Also, if the test is unsure about their demo data fidelity, they can just clear their integration and re-run right? I imagine we'll be "resetting" our integrations periodically anyways to prevent this, along with data buildup.

**Comment by likawind** [comment]
@kenxu95 What you mentioned also makes sense. I'm fine with keeping the current implementation and revisiting once we actually get to managing those integrations.

**Review by likawind — APPROVED** [review]

**Pull request merged**